subtree_oom.go

```diff
@@ -49,25 +50,25 @@ func dmesgContainsOOMKill(line string) bool {
 }
 
 func getPidFromDmesg(line string) (int, bool) {
-	split := strings.Split(line, "Killed process")
-	if len(split) <= 1 {
+	r := regexp.MustCompile(`Killed process (\d+)`)
+	matches := r.FindStringSubmatch(line)
+	if len(matches) != 2 {
```

Comment:
Are we sure there won't be another number that might also match, such that we should keep len(matches) <= 1 instead of != 2?

Reply:
I was thinking it would always return either a slice of length 2 or nil.

The [docs](https://golang.org/pkg/regexp/#Regexp.FindStringSubmatch) say the slice returned will contain the leftmost match ("Killed process 1234") and the matches of its subexpressions ("1234"). I understood that because there's only one subexpression (`(\d+)`) when there is a match it will return a slice of length 2 and when there's no match it will return nil.